src/functional/ffront/parsing/single_static_assign.py

```diff
@@ -65,6 +80,10 @@ def visit_Name(self, node: ast.Name) -> ast.Name:
     class State:
         name_counter: dict[str, int] = field(default_factory=dict)
 
+    @classmethod
+    def mutate_ast(cls, node: ast.AST) -> ast.AST:
```

Comment:
I would standardize the name of the classmethod meant to be used as entry point for all visitor-like classes (parsers, passes, etc.) as `apply()`, as I think it is done right in GT4Py & GTC